tomopy/prep/stripe.py

```diff
@@ -337,3 +348,450 @@ def remove_stripe_sf(tomo, size=5, ncore=None, nchunk=None):
+    Algorithm 3 in the paper. Remove stripes using the sorting technique.
+    Work particularly well for removing partial stripes.
+    Parameters
+    ----------
+    tomo : ndarray
+        3D tomographic data.
+    size : int
+        Window size of the median filter.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs.
+    nchunk : int, optional
+        Chunk size for each core.
+
+    Returns
+    -------
+    ndarray
+        Corrected 3D tomographic data.
+    """
+    arr = mproc.distribute_jobs(
+        tomo,
+        func=_remove_stripe_based_sorting,
+        args=(size,),
+        axis=1,
+        ncore=ncore,
+        nchunk=nchunk)
+    return arr
+
+
+def _create_matindex(nrow, ncol):
+    listindex = np.arange(0.0, ncol, 1.0)
+    matindex = np.tile(listindex,(nrow,1))
+    return matindex
+
+
+def _rs_algorithm_3(sinogram, size, matindex):
+    """
+    Remove stripes using the sorting technique.
+    """
+    sinogram = np.transpose(sinogram)
+    matcomb = np.asarray(np.dstack((matindex, sinogram)))
+    matsort = np.asarray(
+        [row[row[:, 1].argsort()] for row in matcomb])
+    matsort[:, :, 1] = median_filter(matsort[:, :, 1],(size,1))
+    matsortback = np.asarray(
+        [row[row[:, 0].argsort()] for row in matsort])
+    sino_corrected = matsortback[:, :, 1]
+    return np.transpose(sino_corrected)
+
+
+def _remove_stripe_based_sorting(tomo, size):
+    matindex = _create_matindex(tomo.shape[2], tomo.shape[0])
+    for m in range(tomo.shape[1]):
+        sino = tomo[:, m, :]
+        tomo[:, m, :] = _rs_algorithm_3(sino, size, matindex)
+
+ 
+def remove_stripe_based_filtering(
+        tomo, sigma=3, size=21, ncore=None, nchunk=None):
+    """
+    Remove horizontal stripes from sinogram using Nghia Vo's
+    approach :cite:
+    Algorithm 2 in the paper. Remove stripes using the filtering technique.    
+    Parameters
+    ----------
+    tomo : ndarray
+        3D tomographic data.
+    sigma: float
+        Sigma of the Gaussian window which is used to separate 
+        the low-pass and high-pass components of the intensity
+        profiles of each column.
+    size : int
+        Window size of the median filter.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs.
+    nchunk : int, optional
+        Chunk size for each core.
+ 
+    Returns
+    -------
+    ndarray
+        Corrected 3D tomographic data.
+    """
+    arr = mproc.distribute_jobs(
+        tomo,
+        func=_remove_stripe_based_filtering,
+        args=(sigma, size),
+        axis=1,
+        ncore=ncore,
+        nchunk=nchunk)
+    return arr
+
+
+def _create_1d_window(ncol, sigma, pad):
+    window = signal.gaussian(ncol + 2 * pad, std = sigma)
+    return window
+
+
+def _create_listsign(ncol, pad):
+    listsign = np.power(-1.0,np.arange(ncol + 2 * pad))
+    return listsign
+
+ 
+def _rs_algorithm_2(sinogram, window, listsign, sigma, size, pad):
+    """
```

Comment:
Now I realized you have 3 algorithms in the paper. Can we rename them based on the technique rather than the indexing them? We can refer the number of the algorithm in the function docs. I suggest _rs_filter or something like that.

Reply:
Agree